compiler/rustc_trait_selection/src/traits/specialize/specialization_graph.rs

```diff
@@ -226,6 +226,22 @@ fn overlap_check_considering_specialization<'tcx>(
         return Ok(OverlapResult::SpecializeAll(replace_children));
     }
 
+    if overlap_mode.use_negative_impl()
```

Comment:
remove this if, but before you do, let's try to figure out why the `src/test/ui/traits/negative-impls/pin-unsound-issue-66544-derefmut.rs` test gets an error (assuming it does...)

Reply:
probably because we didn't remove the *old* coherence logic